gcc/testsuite/rust.test/compilable/match_self.rs

```diff
@@ -0,0 +1,12 @@
+pub struct Struct;
+
+impl Struct {
+    pub fn method(&mut self) {
+        match self {
```

Comment:
Can you create another test case similar to this but without the match something like:

fn method(&mut self) ->i32 {
  123
}

fn main() {
   let a:Struct;
  a.method()
}

Currently #241 is not implemented yet

Reply:
Sure thing!

Reply:
Without the `match`, the test compiles successfully. I added it in 70ab13c

Reply:
Interesting that it passes lol I guess GCC might be quite permissive but I bet if you run -fdump-tree-gimple and examine the gimple output that the generated method is receiving an instance and not a reference i would have it expected it to fail/crash lol thanks for adding this

Reply:
I will be tackling pattern matching closer to June time with the control flow 2 milestones that I have set out todo just for reference.